wip-0004.md

```diff
@@ -0,0 +1,166 @@
+
+```
+
+                                          Superblock
+
+        superblock_dr_root            superblock_tally_root               ARS_root
+
+      dr_root1       dr_root2       tally_root1       tally_root2     pk1          pk2 
+
+```
+
+Note that verifying data request and tally transactions becomes as simple as providing the valid merkle path to the superblock_dr_root/superblock_tally_root transactions. Identically, proving the membership to the ARS is achieved by again providing a valid merkle path to the ARS_root.
+
+**Specification**
+
+The message to be hashed and signed by the ARS members could look something like this:
+
+```rust
+struct Superblock {
+    struct BlockHeader {
+        struct BlockMerkleRoots {
+            dr_hash_merkle_root: Hash,      
+            tally_hash_merkle_root: Hash,
+            ars_merkle_root: Hash
+        }
+        superblock_beacon: CheckpointBeacon;
+    }
+}
+```
+where *dr_hash_merkle_root* and *tally_hash_merkle_root* are the merkle roots with all dr/tally roots from all blocks that occurred in the last *checkpoints. *ars_merkle_root* refers to the merkle root of the ARS members at the current checkpoint, and *superblock_beacon* refers to the index and the hash of the previous superblock.
+
+The message to be transmitted will be the signature of the hash of the aforementiond superblock:
+
+```rust
+message SuperblockVote {
+    KeyedSignature superblock_sig = 1;
+}
+```
+
+#### Signature aggregation
+The aggregation can happen at two different stages in the protocol:
+
+1. The ARS members append their signature to an aggregated signature they receive. The drawback of this option is that ARS members should verify the aggregated signatures to prevent from aggregating their own signature to an invalid set of signatures.
+
+2. The ARS members just transmit their vote and relay every other ARS members vote. The aggregation this time is in charge of the bridge node, who can easily discard invalid signatures. The drawback of this option is that if the ARS becomes big enough it might flood the network.
+
+
+
+<img src="WIP-0004/sig_aggregation.jpeg"></img>
+
+The proposal implements option 2 due to its similarity with the current way of transmitting blocks every epoch. The mitigations for the potential flooding the network might suffer are still open to discussion and are out of scope for this proposal.
+
+The bridge node in this case will construct the message for the Block Relay. First it will ask the Witnet node it is connected to for the current superblock and the amount of votes it received. Next, it will aggregate those votes into a single aggreated signature. It will also create a merkle proof demonstrating the belonging of each of the voters to the previous superblock ARS. This message will be transmitted to the block relay, which sould verify the merkle proof and implement the GRANDPA protocol.
```

Comment:
```suggestion
The bridge node in this case will construct the message for the Block Relay. First it will ask the Witnet node it is connected to for the current superblock and the amount of votes it received. Next, it will aggregate those votes into a single aggreated signature. It will also create a merkle proof demonstrating the belonging of each of the voters to the previous superblock ARS. This message will be transmitted to the block relay, which should verify the merkle proof and implement the GRANDPA protocol, which implementation will be discussed and presented in another Witnet improval proposal .
```